packages/backend-dynamic-feature-service/src/scanner/schemas.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ScannedPluginPackage } from '@backstage/backend-dynamic-feature-service';
+import { ConfigSchemaPackageEntry } from '@backstage/config-loader';
+import fs from 'fs-extra';
+import * as path from 'path';
+import * as url from 'url';
+import { isEmpty } from 'lodash';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { PackagePlatform, PackageRoles } from '@backstage/cli-node';
+
+export async function gatherDynamicPluginsSchemas(
+  packages: ScannedPluginPackage[],
+  logger: LoggerService,
+  schemaLocator: (platform: PackagePlatform) => string = () =>
+    path.join('dist', 'configSchema.json'),
+): Promise<ConfigSchemaPackageEntry[]> {
+  const allSchemas: { value: any; path: string }[] = [];
+
+  for (const pluginPackage of packages) {
+    const platform = PackageRoles.getRoleInfo(
```

Comment:
I find this to be a bit tricky for frontend plugins, since it relies on them actually being installed in the `app-backend` instance. In the long run that's not gonna be the case right? There might be a couple of plugins brought in from a separate endpoint. Not really something to address as part of this PR, it's more of just another reason for why I want to move away from using config outside of `app.` for the frontend.

Reply:
The dynamic plugin scanner scans for, and discovers, both backend and frontend plugins (even though the loading mechanism is different for frontend and backend).
And since config schemas are still used for frontend plugins, we have to handle both cases here.

> Not really something to address as part of this PR, it's more of just another reason for why I want to move away from using config outside of app. for the frontend.

I get it. :+1: